### PR TITLE
Refactored Multisig benchmarks

### PR DESCRIPTION
**Pull request by maxsam4**
It was not possible to predict `change_all_signers_and_sigs_required` gas cost since we don't know how many signers will be removed. We would have had to take that as input from the user. I removed this function altogether in favor of using the generic `utility.batch`.

### modified API

- `change_all_signers_and_sigs_required` has been removed. The users should use `utility.batch` function.

### CODE REVIEW

File: pallets/multisig/src/lib.rs [hunk 1]
```diff
@@ -299,16 +300,12 @@ decl_module! {
             proposal: Box<T::Proposal>,
             expiry: Option<T::Moment>,
             auto_close: bool
-        ) -> DispatchResultWithPostInfo {
+        ) -> DispatchResult {
```
Comment (Centril) [inline pallets/multisig/src/lib.rs:303]:
```suggestion
        ) {
```

File: pallets/multisig/src/lib.rs [hunk 2]
```diff
@@ -299,16 +300,12 @@ decl_module! {
...
             let sender = ensure_signed(origin)?;
             let sender_did = Context::current_identity_or::<Identity<T>>(&sender)?;
             let sender_signer = Signatory::from(sender_did);
-            let info = PostDispatchInfo {
-                actual_weight: Some(<T as Trait>::WeightInfo::create_or_approve_proposal_as_identity().saturating_add(proposal.get_dispatch_info().weight)),
-                pays_fee: Default::default(),
-            };
             Self::create_or_approve_proposal(multisig, sender_signer, proposal, expiry, auto_close)?;
-            Ok(info)
+            Ok(())
```
Comment (Centril) [inline pallets/multisig/src/lib.rs:308]:
```suggestion
```

**Review by CJP10 — APPROVED** [review]

**Review by adamdossa — APPROVED** [review]

**Review by satyamakgec — APPROVED** [review]

File: pallets/multisig/src/lib.rs [hunk 3]
```diff
@@ -57,11 +57,9 @@
 //! - `add_multisig_signers_via_creator` - Adds a signer to the multisig with the signed being the
 //! creator of the multisig.
 //! - `change_sigs_required` - Changes the number of signatures required to execute a transaction.
-//! - `change_all_signers_and_sigs_required` - Replaces all existing signers of the given multisig
-//! and changes the number of required signatures.
-//! `make_multisig_signer` - Adds a multisig as a signer of the current DID if the current DID is
+//! - `make_multisig_signer` - Adds a multisig as a signer of the current DID if the current DID is
 //! the creator of the multisig.
-//! `make_multisig_primary` - Adds a multisig as the primary key of the current DID if the current did
+//! - `make_multisig_primary` - Adds a multisig as the primary key of the current DID if the current did
```
Comment (vkomenda) [inline pallets/multisig/src/lib.rs:62]:
```suggestion
//! - `make_multisig_primary` - Adds a multisig as the primary key of the current DID if the current DID
```

File: pallets/multisig/src/lib.rs [hunk 4]
```diff
@@ -945,24 +889,27 @@ impl<T: Trait> Module<T> {
                     );
                 }
                 if proposal_details.approvals >= Self::ms_signs_required(&multisig) {
```
Comment (vkomenda) [inline pallets/multisig/src/lib.rs:891]: Indentation could be reduced if we return here:
```
if proposal_details.approvals < Self::ms_signs_required(&multisig) {
    return Ok(());
}
...
```

Comment (maxsam4) [reply pallets/multisig/src/lib.rs:891]: @vkomenda This code is inside a `match`. Returning from here will return from the full function, not just the match. Unfortunately, rust doesn't allow breaking from a match without ugly hacks.

Comment (Centril) [reply pallets/multisig/src/lib.rs:891]: In a situation like this I'd instead opt to move the stuff in the `if`'s then block into a function

Comment (maxsam4) [reply pallets/multisig/src/lib.rs:891]: Yeah, we can do it in a future refactoring maybe. We can live with some indentation for now :)

**Pull request merged**